src/ResultField/BeezUp.php

```diff
@@ -107,78 +76,39 @@ public function generateResultFields(array $formatSettings = []):array
 			$barcodeMutator->addMarket($reference);
 		}
 
-        $fields = [
-            [
-                //item
-                'item.id',
-                'item.manufacturer.id',
-
-                //variation
-                'id',
-                'variation.availability.id',
-                'variation.stockLimitation',
-                'variation.weightG',
-                'variation.model',
-                'variation.number',
-
-                //attributes
-                'attributes.valueId',
-
-                //images
-                'images.item.urlMiddle',
-                'images.item.urlPreview',
-                'images.item.urlSecondPreview',
-                'images.item.url',
-                'images.item.position',
-                'images.item.path',
-
-                'images.variation.urlMiddle',
-                'images.variation.urlPreview',
-                'images.variation.urlSecondPreview',
-                'images.variation.url',
-                'images.variation.position',
-                'images.variation.path',
-
-                //defaultCategories
-                'defaultCategories.id',
-
-                //unit
-                'unit.content',
-                'unit.id',
-
-                //barcodes
-                'barcodes.code',
-                'barcodes.id',
-                'barcodes.type',
-
-                //properties
-				'properties.property.id',
-				'properties.property.valueType',
-				'properties.selection.name',
-				'properties.selection.lang',
-				'properties.texts.value',
-				'properties.texts.lang',
-				'properties.valueInt',
-				'properties.valueFloat',
-
-			],
+        /**
+         * @var ResultFieldDataProvider $resultFieldHelper
+         */
+        $resultFieldHelper = pluginApp(ResultFieldDataProvider::class);
+        if($resultFieldHelper instanceof ResultFieldDataProvider)
+        {
+            $resultFields = $resultFieldHelper->getResultFields($settings);
+        }
 
-            [
+        if(isset($resultFields) && is_array($resultFields) && count($resultFields))
+        {
+            $fields[0] = $resultFields;
+            $fields[1] = [
                 $languageMutator,
                 $defaultCategoryMutator,
-				$barcodeMutator,
+                $barcodeMutator,
                 $keyMutator
-            ],
-        ];
+            ];
 
-        if($reference != -1)
+            if($reference != -1)
+            {
+                $fields[1][] = $imageMutator;
+            }
+        }
+        else
         {
-            $fields[1][] = $imageMutator;
+            $this->getLogger(__METHOD__)->critical('ElasticExportBeezUp::log.resultFieldError');
+            exit();
         }
 
-        foreach($itemDescriptionFields as $itemDescriptionField)
+        if($reference != -1)
```

Comment:
I think that here, you're adding for the second time the imageMutator. The ES will return an empty array from what I could test it.

Reply:
fixed